tests/test_gpt_cli.py

```diff
@@ -1,19 +1,68 @@
-"""Pytest for ask_gpt/main.py"""
+"""Pytest for gpt_review/main.py"""
+from dataclasses import dataclass
 import os
 import pytest
 import subprocess
 import sys
 
 from gpt_review._gpt_cli import cli
+from knack.util import CLIError
+
+
+@dataclass
+class TestCase:
+    command: str
+    expected_error: CLIError = None
+    expected_error_message: str = ""
+
 
 ROOT_COMMANDS = [
-    "--version",
-    "--help",
+    TestCase("--version"),
+    TestCase("--help"),
 ]
 
 ASK_COMMANDS = [
-    "ask --help",
-    "ask how are you",
+    TestCase("ask --help"),
+    TestCase("ask how are you"),
+    TestCase("ask how are you --max-tokens=5"),
+    TestCase(
+        "ask how are you --max-tokens 0", CLIError, "b'ERROR: --max_tokens must be an integer between 1 and 4000\\n'"
+    ),
+    TestCase("ask how are you --temperature 1"),
+    TestCase("ask how are you -t 0.5"),
+    TestCase("ask how are you -t 9", CLIError, "b'ERROR: --temperature must be a float between 0 and 1\\n'"),
+    TestCase("ask how are you --top-p 0.7"),
+    TestCase("ask how are you --top-p 4.5", CLIError, "b'ERROR: --top-p must be a float between 0 and 1\\n'"),
+    TestCase("ask how are you --frequency-penalty 1"),
+    TestCase(
+        "ask how are you --frequency-penalty 3",
+        CLIError,
+        "b'ERROR: --frequency-penalty must be a float between 0 and 2\\n'",
+    ),
+    TestCase("ask how are you --presence-penalty 0.7"),
+    TestCase(
+        "ask how are you --presence-penalty 8.7",
+        CLIError,
+        "b'ERROR: --presence-penalty must be a float between 0 and 2\\n'",
+    ),
+    TestCase("ask how are you --max-tokens=5 --temperature 0.1"),
+    TestCase(
+        "ask how are you --max-tokens 5 -t 8", CLIError, "b'ERROR: --temperature must be a float between 0 and 1\\n'"
+    ),
+    TestCase("ask how are you --max-tokens=5 --top-p 0.7"),
+    TestCase(
+        "ask how are you --max-tokens 5000 --top-p 0.7",
+        CLIError,
+        "b'ERROR: --max_tokens must be an integer between 1 and 4000\\n'",
+    ),
+    TestCase("ask how are you --frequency-penalty 1 --presence-penalty 0.7"),
+    TestCase(
+        "ask how are you --frequency-penalty 4 --presence-penalty 4",
+        CLIError,
+        "b'ERROR: --frequency-penalty must be a float between 0 and 2\\n'",
```

Comment:
instead of hard coding the min and max both here, and in the code, try extracting it into a constant that is called in both places.